File: DTO/Models/Address.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DTO.Models
{
    public class Address
    {
        public int Id { get; set; }
        public string Street { get; set; }
        public string Number { get; set; }
        public City City { get; set; }        
        public Address()
        {

        }
        public Address(int id, string street, string number, City city)
        {
            Id = id;
            Street = street;
            Number = number;
            City = city;
        }
 
    }
}

File: Angular-Project/WineSellingProject/src/app/Models/Address.ts
import { IAddress } from "../interfaces/IAddress";
import { City } from "./City";

export class Address implements IAddress{
    
    Id: number;
    Street: string;
    Number: number;
    City: City;

 
    constructor(Id: number, Street: string, number: number, City: City) {
        this.Id = Id;
        this.Street = Street;
        this.Number = number;
        this.City = City;
    }
    
}
